### ISSUE TITLE
Signature mismatch using proxy

### ISSUE THREAD

**Issue by matthiasbe** [woothemes/wc-api-node#14]
Hi,
First I'd like to tell you how grateful I am to find such a practical API adapted on NodeJS.
I am currently using it under Ionic to fetch informations from the website designed with wordpress/woocommerce. I had an issue recently with mismatching signature on OAuth. I've managed to print both non-hashed signature and I found out the problem.

As I am currently testing on browser I had to avoid CORS, and this was done with a proxy thank to ionic. But now I can use API on debug because signatures are built with the proxy address (localhost) on the client side, whereas the serve is building them with the server address (example.com).

In my opinion, the best solution would be to take proxy in the parameters when declaring the wc-api class.

**Comment by claudiosmweb** [comment]
Fell free to submit a Pull Request with what you need to make it work :)

**Comment by matthiasbe** [comment]
Tried with this minimal change. https://github.com/matthiasbe/wc-api-node/commit/6fc347659cf227b7074f28e3e0d5a3ab463130fa

I declare the API this way

```javascript
var wc = new WooCommerceAPI({
  proxy: 'http://mysite.com/wc-api/v3/products',
  consumerKey: 'ck_xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx',
  consumerSecret: 'cs_xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx',
  url: 'http://localhost:8100/rest-api/'
});
```

No more errors but now the response is empty. Screenshot from firebug :
![screenshot_null_response](https://cloud.githubusercontent.com/assets/9254232/13763221/e4efa9c0-ea44-11e5-89ff-18ac66112eeb.png)

If you have an idea, or maybe I'm just taking your code completely wrong...

**Comment by matthiasbe** [comment]
I was actually checking `err` variable instead of `data`.
I think then that it is working, I have just something to add and I'll submit a Pull Request.

**Comment by claudiosmweb** [comment]
> I think then that it is working, I have just something to add and I'll submit a Pull Request.

?

**Comment by matthiasbe** [comment]
> ?
I was saying the code wasn't ready to be pull-requested I had to make some more changes.

**Comment by claudiosmweb** [comment]
Thanks :)

**Comment by claudiosmweb** [comment]
Closed in favor of #15

**Comment by claudiosmweb** [comment]
I made some changes in https://github.com/woothemes/wc-api-node/commit/cf707822b5b94b6af21565b13afdb4b4332507ce

You can update to 1.1.1 now.

Thanks for help us :)
